service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/NonConsultationResourceMapper.java

```diff
@@ -254,4 +277,17 @@ private boolean isIgnoredResource(Resource resource) {
 
         return false;
     }
+
+    private boolean isEndedAllergyList(Resource resource) {
+        if (resource.getResourceType().equals(ResourceType.List)) {
+            var list = (ListResource) resource;
+            if (list.hasCode()
```

Comment:
Instead of those checks inside if statement, I would use Optional.ofNullable:
```suggestion
            var endedAllergies = Optional.ofNullable(list.getCode())
                .map(CodeableConcept::getCodingFirstRep)
                .filter(coding -> coding.getCode().equals(ENDED_ALLERGIES_CODE))
                .orElse(null);
            if (endedAllergies != null) {
                return list.hasContained();
            }
```

Reply:
Nice :) Implemented this